zerver/lib/test_classes.py

```diff
@@ -339,79 +341,143 @@ def client_patch_multipart(
+        return django_client.put(
+            url, data=data, content_type="application/json", follow=follow, secure=secure, **extra
+        )
 
     @instrument_url
     def client_delete(
-        self, url: str, info: Dict[str, Any] = {}, **kwargs: ClientArg
+        self,
+        url: str,
+        info: Dict[str, Any] = {},
+        skip_user_agent: bool = False,
+        follow: bool = False,
+        secure: bool = False,
+        intentionally_undocumented: bool = False,
+        **extra: str,
     ) -> "TestHttpResponse":
         encoded = urllib.parse.urlencode(info)
-        kwargs["content_type"] = "application/x-www-form-urlencoded"
+        extra["content_type"] = "application/x-www-form-urlencoded"
         django_client = self.client  # see WRAPPER_COMMENT
-        self.set_http_headers(kwargs)
-        result = django_client.delete(url, encoded, **kwargs)
-        self.validate_api_response_openapi(url, "delete", result, info, kwargs)
+        self.set_http_headers(extra, skip_user_agent)
+        result = django_client.delete(url, encoded, follow=follow, secure=secure, **extra)
+        self.validate_api_response_openapi(
+            url,
+            "delete",
+            result,
+            info,
+            extra,
+            intentionally_undocumented=intentionally_undocumented,
+        )
         return result
 
     @instrument_url
     def client_options(
-        self, url: str, info: Dict[str, Any] = {}, **kwargs: ClientArg
+        self,
+        url: str,
+        info: Dict[str, Any] = {},
+        skip_user_agent: bool = False,
+        follow: bool = False,
+        secure: bool = False,
+        **extra: str,
     ) -> "TestHttpResponse":
         django_client = self.client  # see WRAPPER_COMMENT
-        self.set_http_headers(kwargs)
-        return django_client.options(url, info, **kwargs)
+        self.set_http_headers(extra, skip_user_agent)
+        return django_client.options(url, info, follow=follow, secure=secure, **extra)
 
     @instrument_url
     def client_head(
-        self, url: str, info: Dict[str, Any] = {}, **kwargs: ClientArg
+        self,
+        url: str,
+        info: Dict[str, Any] = {},
+        skip_user_agent: bool = False,
+        follow: bool = False,
+        secure: bool = False,
+        **extra: str,
     ) -> "TestHttpResponse":
         django_client = self.client  # see WRAPPER_COMMENT
-        self.set_http_headers(kwargs)
-        return django_client.head(url, info, **kwargs)
+        self.set_http_headers(extra, skip_user_agent)
+        return django_client.head(url, info, follow=follow, secure=secure, **extra)
 
     @instrument_url
     def client_post(
         self,
         url: str,
         info: Union[str, bytes, Dict[str, Any]] = {},
-        **kwargs: ClientArg,
+        skip_user_agent: bool = False,
+        follow: bool = False,
+        secure: bool = False,
+        intentionally_undocumented: bool = False,
+        **extra: str,
```

Comment:
@PIG208 , I'm writing tests for a hook, which needs to pass extra headers in client_post, but here `**extra` is type of `str` so there is mypy lint error on passing headers as`Dict[str,str]`. Because `django_client.post()` accepts headers dict, is there any specific reason for this ?

Reply:
For now I got a workaround, I passed header directly and used request.META.get() in the view.

Reply:
I'm not sure if I understand your question. Are you passing headers like this?

```
headers = {"HEADER_A": "str", "HEADER_B": "str"}
client_post(..., extra=headers)
```

Generally, we pass extra headers as kwargs, so you can do things like

```
client_post(.., HEADER_A="str", HEADER_B="str)
```

Reply:
No, I was passing as 
```
headers = {"Hook-Name": "str"}
client_post(..., headers=headers)
```
Got it now. It was working fine while running the tests. mypy tests failed, that's how I got it. 

> I passed header directly and used request.META.get() in the view.

I meant as kwargs. I can also access using `request.headers.get()`, didn't try earlier.

I got it now. Thank you !

Reply:
Oh, so `client_post(..., **headers)` should be what you were looking for. Glad that it helps!

Reply:
For a bit more context @brijsiyag , here is the discussion that motivated this type annotation: https://chat.zulip.org/#narrow/stream/3-backend/topic/Typing.20keyword.20variadic.20arguments.20for.20test.20client.20helpers